legate/core/shape.py

```diff
@@ -148,55 +144,95 @@ def __eq__(self, other: object) -> bool:
             else:
                 return self.extents == other.extents
         elif isinstance(other, (int, Iterable)):
-            lh = _cast_tuple(self, self.ndim)
-            rh = _cast_tuple(other, self.ndim)
+            lh = self.extents
+            rh = (
```

Comment:
Can we roll the `other.extents if isinstance(other, Shape) else ...` part into `_cast_tuple` as well?

Reply:
That's the way it was, I was trying to avoid calling `_cast_tuple` at all when possible